draft-ietf-tls-esni.md

```diff
@@ -801,20 +785,35 @@ of the "ech_required" alert to detect this case.
 
 ### HelloRetryRequest {#server-hrr}
 
-In case a HelloRetryRequest (HRR) is sent, the client-facing server MUST
-consistently accept or decline ECH between the two ClientHellos, using the same
-ECHConfig, and abort the handshake if this is not possible. This is achieved as
-follows. Let CH1 and CH2 denote, respectively, the first and second ClientHello
-transmitted on the wire by the client:
-
-1. If CH1 contains the "encrypted_client_hello" extension but CH2 does not, or
-   if CH2 contains the "encrypted_client_hello" extension but CH1 does not, then
-   the server MUST abort the handshake with an "illegal_parameter" alert.
-1. If the "encrypted_client_hello" extension is sent in CH2, the server follows
-   the procedure in {{client-facing-server}} to decrypt the extension, but it
-   uses the previously-selected ECHConfig as the set of candidate ECHConfigs.
-   If decryption fails, the server aborts the connection with a "decrypt_error"
-   alert rather than continuing the handshake with the second ClientHelloOuter.
+After sending a HelloRetryRequest, the client-facing server does not repeat the
```

Comment:
Self-review: "After sending a HelloRetryRequest" is slightly odd because, on ECH accept, the client-facing server forwards the HRR from the backend server. At the same time, this section also talks about ECH reject, since we want to clarify that you don't process ECH on the second ClientHello as usual.

Perhaps: "After sending or forwarding a HelloRetryRequest"?

Reply:
That sounds good to me.